src/ripple/app/tx/applySteps.h

```diff
@@ -47,7 +91,7 @@ struct PreflightResult
     beast::Journal const j;
 
     /// Intermediate transaction result
-    TER const ter;
+    NotTEC const ter;
 
     /// Constructor
     template<class Context>
```

Comment:
On line 104 here, we're initializing a variable of type `NotTEC` with a variable of type `TER`. This seems like a bug, I'd want the type system to catch this (TER could be a TEC, right?)

Reply:
Yup.  That looks like a bug.  I'd have expected a compile error out of that.  Investigating.

Reply:
Too bad it's a PITA to write a unit test for a compile error :-) It's easy to see how this could slip through.

Reply:
There's a reason I haven't written the unit tests yet 😀.  I'm poking at it right now.  I'll let you know what I find (assuming I figure it out).

Reply:
Actually, we should be able to use something like `static_assert(!is_convertible` now that I think about it.

Reply:
Found it.  It's the
```
constexpr explicit TERSubset (int rhs) : code_ (rhs) { } // For TxMeta
```
Apparently `explicit` wasn't enough to keep that constructor from being called.  For now I'm changing it to a named static factory function so the compiler won't be so helpful.  Good catch!  Thanks.